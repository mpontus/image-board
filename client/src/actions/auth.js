export const AUTHENTICATED = "AUTHENTICATED";
export const LOGOUT = "LOGOUT";

export const authenticated = idToken => ({
  type: AUTHENTICATED,
  payload: {
    idToken
  }
});

export const logout = () => ({
  type: LOGOUT
});
