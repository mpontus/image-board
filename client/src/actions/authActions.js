export const LOGIN = "LOGIN";
export const AUTHENTICATED = "AUTHENTICATED";
export const LOGOUT = "LOGOUT";

export const login = () => ({
  type: LOGIN
});

export const authenticated = idToken => ({
  type: AUTHENTICATED,
  payload: {
    idToken
  }
});

export const logout = () => ({
  type: LOGOUT
});
