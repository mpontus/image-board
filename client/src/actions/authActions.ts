export const AUTHENTICATED = "AUTHENTICATED";
export const LOGIN = "LOGIN";
export const LOGOUT = "LOGOUT";

export type Action =
  | {
      type: typeof AUTHENTICATED;
      payload: {
        idToken: string;
      };
    }
  | {
      type: typeof LOGIN;
    }
  | {
      type: typeof LOGOUT;
    };

export const authenticated = (idToken: string): Action => ({
  type: AUTHENTICATED,
  payload: {
    idToken,
  },
});

export const login = (): Action => ({
  type: LOGIN,
});

export const logout = (): Action => ({
  type: LOGOUT,
});
