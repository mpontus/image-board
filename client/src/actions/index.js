export {
  FETCH_POSTS,
  FETCH_POSTS_SUCCESS,
  FETCH_POSTS_ERROR,
  CREATE_POST,
  CREATE_POST_SUCCESS,
  CREATE_POST_ERROR,
  LIKE_POST,
  LIKE_POST_SUCCESS,
  LIKE_POST_ERROR,
  DELETE_POST,
  DELETE_POST_SUCCESS,
  DELETE_POST_ERROR,
  UPLOAD_PROGRESS,
  fetchPosts,
  fetchPostsSuccess,
  fetchPostsError,
  createPost,
  createPostSuccess,
  createPostError,
  likePost,
  likePostSuccess,
  likePostError,
  deletePost,
  deletePostSuccess,
  deletePostError,
  uploadProgress
} from "./postsActions";

export {
  LOGIN,
  AUTHENTICATED,
  LOGOUT,
  login,
  authenticated,
  logout
} from "./authActions";
