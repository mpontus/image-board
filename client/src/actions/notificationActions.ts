export const DISMISS_NOTIFICATION = "DISMISS_NOTIFICATION";

export const dismissNotification = () => ({
  type: DISMISS_NOTIFICATION,
});
