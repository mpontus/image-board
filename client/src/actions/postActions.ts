import nanoid from "nanoid";
import { PageResponse } from "../api";
import { Post, User } from "../models";
import { fileToDataUrl, getImageDimensions } from "../utils";

export const LOAD_POSTS = "LOAD_POSTS";
export const LOAD_POSTS_REJECT = "LOAD_POSTS_REJECT";
export const LOAD_POSTS_RESOLVE = "LOAD_POSTS_RESOLVE";
export const CREATE_POST = "CREATE_POST";
export const CREATE_POST_RESOLVE = "CREATE_POST_RESOLVE";
export const CREATE_POST_REJECT = "CREATE_POST_REJECT";
export const UPLOAD_PROGRESS = "UPLOAD_PROGRESS";

export type Action =
  | {
      type: typeof LOAD_POSTS;
    }
  | {
      type: typeof LOAD_POSTS_RESOLVE;
      payload: {
        total: number;
        posts: Post[];
      };
    }
  | {
      type: typeof LOAD_POSTS_REJECT;
      payload: {
        error: Error;
      };
    }
  | {
      type: typeof CREATE_POST;
      payload: {
        file: File;
        post: Post;
      };
    }
  | {
      type: typeof CREATE_POST_RESOLVE;
      payload: {
        post: Post;
        instance: Post;
      };
    }
  | {
      type: typeof CREATE_POST_REJECT;
      payload: {
        post: Post;
        error: Error;
      };
    }
  | {
      type: typeof UPLOAD_PROGRESS;
      payload: {
        key: string;
        bytesTransferred: number;
        bytesTotal: number;
      };
    };

export const loadPosts = (): Action => ({
  type: LOAD_POSTS
});

export const loadPostsResolve = ({ total, items }: PageResponse): Action => ({
  type: LOAD_POSTS_RESOLVE,
  payload: {
    total,
    posts: items
  }
});

export const loadPostsReject = (error: Error): Action => ({
  type: LOAD_POSTS_REJECT,
  payload: {
    error
  }
});

export const createPost = (file: File, user: User): Promise<Action> =>
  Promise.all([fileToDataUrl(file), getImageDimensions(file)]).then(
    ([url, { width, height }]): Action => ({
      type: CREATE_POST,
      payload: {
        file,
        post: {
          id: `temp/${nanoid()}`,
          author: user,
          picture: {
            url,
            width,
            height
          },
          likesCount: 1,
          isLiked: true,
          timestamp: Date.now()
        }
      }
    })
  );

export const createPostResolve = (post: Post, instance: Post): Action => ({
  type: CREATE_POST_RESOLVE,
  payload: {
    post,
    instance
  }
});

export const createPostReject = (post: Post, error: Error): Action => ({
  type: CREATE_POST_REJECT,
  payload: {
    post,
    error
  }
});

export const uploadProgress = (
  key: string,
  bytesTransferred: number,
  bytesTotal: number
): Action => ({
  type: UPLOAD_PROGRESS,
  payload: {
    key,
    bytesTransferred,
    bytesTotal
  }
});
