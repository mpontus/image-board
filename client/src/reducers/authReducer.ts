import { AUTHENTICATED } from "../actions";

const initialState = null;

export default (state = initialState, action) => {
  switch (action.type) {
    case AUTHENTICATED: {
      const { idToken } = action.payload;

      return {
        token: idToken,
      };
    }

    default: {
      return state;
    }
  }
};
