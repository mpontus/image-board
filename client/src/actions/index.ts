import { Action as PostAction } from "./postActions";
import { Action as AuthAction } from "./authActions";

export type Action = PostAction | AuthAction;

export {
  LOAD_POSTS,
  LOAD_POSTS_RESOLVE,
  LOAD_POSTS_REJECT,
  CREATE_POST,
  CREATE_POST_RESOLVE,
  CREATE_POST_REJECT,
  loadPosts,
  loadPostsResolve,
  loadPostsReject,
  createPost,
  createPostResolve,
  createPostReject,
} from "./postActions";

export {
  AUTHENTICATED,
  LOGIN,
  LOGOUT,
  authenticated,
  login,
  logout,
} from "./authActions";
