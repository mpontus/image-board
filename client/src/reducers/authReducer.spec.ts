import authReducer from "./authReducer";
import { authenticated } from "../actions";

const initialState = authReducer(undefined, {} as any);

describe("authReducer", () => {
  describe("initial state", () => {
    it("should match snapshot", () => {
      expect(initialState).toEqual({
        idToken: null
      });
    });
  });

  describe("reflecting authentication status", () => {
    it("should store id token in the state", () => {
      const idToken = "foo";
      const action = authenticated(idToken);
      const state = authReducer(initialState, action);

      expect(state.idToken).toBe(idToken);
    });
  });
});
