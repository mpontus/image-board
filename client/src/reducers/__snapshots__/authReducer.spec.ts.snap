// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authReducer initial state should match snapshot 1`] = `
Object {
  "idToken": null,
}
`;
